tests/unit/code_provider/test_repo_name_validation.py

```
@@ -0,0 +1,216 @@
+        with patch.object(GithubService, "__init__", lambda self, db: None):
+            svc = GithubService.__new__(GithubService)
+            return svc
+
+    @pytest.mark.asyncio
+    async def test_rejects_invalid_repo_name(self, service):
+        with pytest.raises(ValueError, match="Invalid repository name format"):
+            await service.check_public_repo("not-a-valid-name")
+
+    @pytest.mark.asyncio
+    async def test_rejects_repo_ending_with_dot_git(self, service):
+        with pytest.raises(ValueError, match="end with '.git'"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)